src/main/java/io/debezium/connector/jdbc/HibernateRecordParser.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.jdbc;
+
+import static io.debezium.connector.jdbc.HibernateJAXBRecordMapper.DEBEZIUM_ID;
+import static io.debezium.connector.jdbc.HibernateJAXBRecordMapper.getHibernateAttributeType;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.hibernate.type.BigIntegerType;
+import org.hibernate.type.BooleanType;
+import org.hibernate.type.ByteType;
+import org.hibernate.type.FloatType;
+import org.hibernate.type.IntegerType;
+import org.hibernate.type.LongType;
+import org.hibernate.type.TimeType;
+import org.hibernate.type.TimestampType;
+
+public class HibernateRecordParser {
+    private static final String DEBEZIUM_DATATYPE_PREFIX = "io.debezium";
+
+    /** Create dataMap from field name to its data from {@link SinkRecord} with record Data Maps made by
+     *
+     * @param record
+     * @return
+     */
+    public static Map<String, Object> getDataMap(SinkRecord record) throws Exception {
+        Map<String, Object> dataMap = new HashMap<>();
+
+        Struct value = (Struct) record.value();
+        Schema schema = record.valueSchema();
+        for (Field field : schema.fields()) {
+            String stringData = "";
+
+            if (field.schema().type() == Schema.Type.STRUCT) {
+                stringData = getJsonString((Struct) value.get(field));
```

Comment:
So let's avoid conversion into Strings if possible (e.g. numeric values should be propagated as numbers). I also don't see where this extracts values from the "after" part of change events specifically?